### PR TITLE
librbd: allow to open an image without opening the parent image

### PR DESCRIPTION
**Pull request by rjfd**
Fixes: http://tracker.ceph.com/issues/18325

Signed-off-by: Ricardo Dias <rdias@suse.com>

### CODE REVIEW

File: src/librbd/ImageState.h [hunk 1]
```diff
@@ -25,8 +25,8 @@ class ImageState {
   ImageState(ImageCtxT *image_ctx);
   ~ImageState();
 
-  int open();
-  void open(Context *on_finish);
+  int open(bool dont_open_parent=false);
```
Comment (dillaman) [inline src/librbd/ImageState.h]: Nit: can we rename this to something like ```skip_open_parent``` everywhere? Something about "dont" feels a bit off

Comment (dillaman) [inline src/librbd/ImageState.h]: Nit: avoid defaulting the parameter here and below

File: src/librbd/ImageState.h [hunk 2]
```diff
@@ -25,8 +25,8 @@ class ImageState {
...
+  void open(Context *on_finish, bool dont_open_parent=false);
```
Comment (dillaman) [inline src/librbd/ImageState.h]: Nit: can you move ```on_finish``` to be the last param?

File: src/librbd/image/OpenRequest.h [hunk 3]
```diff
@@ -20,7 +20,12 @@ template <typename ImageCtxT = ImageCtx>
 class OpenRequest {
 public:
   static OpenRequest *create(ImageCtxT *image_ctx, Context *on_finish) {
```
Comment (dillaman) [inline src/librbd/image/OpenRequest.h]: Nit: only need one factory method -- remove this one and update where necessary

File: src/librbd/image/RefreshRequest.h [hunk 4]
```diff
@@ -28,10 +28,17 @@ class RefreshRequest {
 public:
   static RefreshRequest *create(ImageCtxT &image_ctx, bool acquiring_lock,
```
Comment (dillaman) [inline src/librbd/image/RefreshRequest.h]: Nit: same comment here re: only having one factory method

**Comment by rjfd** [comment]
@dillaman updated PR with your comments

**Comment by rjfd** [comment]
jenkins retest this please

File: src/librbd/ImageState.h [hunk 5]
```diff
@@ -26,7 +26,9 @@ class ImageState {
   ~ImageState();
 
   int open();
+  int open(bool skip_open_parent);
```
Comment (dillaman) [inline src/librbd/ImageState.h]: Nit: can you just combine these methods and fix the callsites that now need to pass "false"?

**Comment by rjfd** [comment]
@dillaman pushed new changes

**Comment by rjfd** [comment]
@dillaman done

**Pull request merged**